models/DataObject/ClassDefinition/CustomLayout.php

```diff
@@ -184,6 +196,102 @@ public function save()
         }
     }
 
+
+    private function saveCustomLayoutFile($saveDefinitionFile = true)
+    {
+        // save definition as a php file
+        $definitionFile = $this->getDefinitionFile();
+        if (!is_writable(dirname($definitionFile)) || (is_file($definitionFile) && !is_writable($definitionFile))) {
+            throw new \Exception(
+                'Cannot write definition file in: '.$definitionFile.' please check write permission on this directory.'
+            );
+        }
+
+        $infoDocBlock = $this->getInfoDocBlock();
+
+        $clone = clone $this;
+        $clone->setDao(null);
+        unset($clone->fieldDefinitions);
+
+        self::cleanupForExport($clone->layoutDefinitions);
+
+        if ($saveDefinitionFile) {
+            $exportedCustomLayout = var_export($clone, true);
```

Comment:
there is a pimcore function that does exactly that:

```
use Pimcore\File;

$contents = to_php_data_file_format($this->data);
File::putPhpFile($this->filePath, $contents);
```

Reply:
yup! currently it is not possible to attach comments to file with "to_php_data_file_format" method.. may be we can change it to accept comments also ;)

Reply:
perfect 👍